Libraries/Opc.Ua.Client/SessionReconnectHandler.cs

```diff
@@ -38,13 +38,66 @@ namespace Opc.Ua.Client
     /// </summary>
     public class SessionReconnectHandler : IDisposable
     {
+        /// <summary>
+        /// The minimum reconnect period in ms.
+        /// </summary>
+        public const int MinReconnectPeriod = 500;
+
+        /// <summary>
+        /// The maximum reconnect period in ms.
+        /// </summary>
+        public const int MaxReconnectPeriod = 30000;
+
+        /// <summary>
+        /// The default reconnect period in ms.
+        /// </summary>
+        public const int DefaultReconnectPeriod = 1000;
+
+        /// <summary>
+        /// The internal state of the reconnect handler.
+        /// </summary>
+        public enum ReconnectState
+        {
+            /// <summary>
+            /// The reconnect handler is ready to start the reconnect timer.
+            /// </summary>
+            Ready = 0,
+
+            /// <summary>
+            /// The reconnect timer is triggered and waiting to reconnect.
+            /// </summary>
+            Triggered = 1,
+
+            /// <summary>
+            /// The reconnection is in progress.
+            /// </summary>
+            Reconnecting = 2,
+
+            /// <summary>
+            /// The reconnect handler is disposed and can not be used for further reconnect attempts.
+            /// </summary>
+            Disposed = 4
+        };
+
         /// <summary>
         /// Create a reconnect handler.
         /// </summary>
         /// <param name="reconnectAbort">Set to <c>true</c> to allow reconnect abort if keep alive recovered.</param>
-        public SessionReconnectHandler(bool reconnectAbort = false)
+        /// <param name="maxReconnectPeriod">
+        ///     The upper limit for the reconnect period after exponential backoff.
+        ///     -1 (default) indicates that no exponential backoff should be used.
+        /// </param>
+        public SessionReconnectHandler(bool reconnectAbort = false, int maxReconnectPeriod = -1)
```

Comment:
Why does maxReconnectPeriod not default to MaxReconnectPeriod Or why do we not have a const for NoExponentialBackoffPeriod defined and used?

Reply:
to make it backward compatible, defaults reflect legacy behavior